### DISCUSSION TITLE
Cancellation support for createResource

### DISCUSSION THREAD

**Discussion by hackwaly** [solidjs/solid#1258] [Ideas]
Use case: Auto cancel ajax request on dispose.

**Comment by atk** [comment]
There is @solid-primitives/fetch that allows you to make a request abortable via its composable `withAbort` modifier. If used inside a reactive context, it will already auto-abort.

**Comment by ryansolid** [comment]
This has always confused me a little. Isn't it too late to cancel at that point? Like the request has gone out. You don't want it to affect your app, but like that ship has sailed.

**Reply by hackwaly** [reply]
This can save a lot server resources any way.

**Comment by NunoCruzSW** [comment]
Not sure if it's too late to add something to this discussion, but here's a scenario that might help illustrate the issue.

Imagine the server API is very slow — the initial request takes several seconds to complete. During that time, the user types something or changes a filter, which triggers a new API call. Both requests reach the server, and both eventually respond, but the first one takes longer to process on the client side. As a result, the first (older) response may arrive after the second (newer) one, causing the UI to show outdated data — the opposite of what the user expects.

Ideally, we should be able to cancel or abort previous requests so that outdated responses don't overwrite the results of the most recent request.

Of curse this only makes sense if the promise is also cancelable, or has a way to reject previous resolves

**Comment by atk** [comment]
We now have `makeAbortable` in `@solid-primitives/resource` - it's very lightweight and easily composable.
